py/foundation/entity.py

```diff
@@ -27,6 +28,7 @@
 
 
 class Entity(abc.ABC, Generic[E]):
+  bbox: BBox
```

Comment:
I don't actually know the best way to enforce the contract that derived classes from `Entity` must have a `bbox` attribute. It seems if I omit it in a derived class definition, mypy won't complain. Using `@property` causes problems with dataclass however.

Reply:
I don't think it's that important to enforce technically. It's enough to enforce it socially.